File: cosmos/Cosmos-API/src/main/java/com/planbtech/cosmos/model/entites/Gender.java
package com.planbtech.cosmos.model.entites;

import com.planbtech.cosmos.model.Interfaces.IGender;
import jakarta.persistence.*;

@Entity(name = "gender")
public class Gender implements IGender {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "gender_id")
    private Long id;

    private String gender;

    @Override
    public Long getId() {
        return this.id;
    }

    @Override
    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public String getGender() {
        return this.gender;
    }

    @Override
    public void setGender(String gender) {
        this.gender = gender;
    }
}

File: cosmos/ProConnect/src/app/Class/gender.ts
import { IGender } from "../interfaces/igender";

export class Gender implements IGender{
  gender: string = '';
  id:number = 0;

  getGender(): string {    return this.gender;}
  getId(): number {    return this.id;}

  setId(id: number): void {    this.id = id;}
  setGender(gender: string): void {    this.gender = gender;}

  verificagenero():boolean{
    if(this.gender != "" && this.id != 0){
      return true;
    }
    return false
  }
}
